### PR TITLE
Updated to work with vn3.0.5 of the cf-checker

### PR DESCRIPTION
**Pull request by agstephens**
Version number is at 3.0.1.

 - Dependency on CDAT (cdms) has been removed.
 - Added dependence on `netCDF4` and `cfunits` libraries.

Please pull this change to conda-forge and test it integrates with other packages.

### CODE REVIEW

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml]: Should be lower case.

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml]: This package does not exist on conda-forge. You'll need to submit it first to staged-recipes.

**Review by ocefpaf — CHANGES_REQUESTED** [review]

**Comment by agstephens** [comment]
Now updated to work with v3.0.5 of cfchecker (latest release and in PyPI).

**Comment by ocefpaf** [comment]
You need to update the checksum in https://github.com/conda-forge/cfchecker-feedstock/pull/5/files#diff-e178b687b10a71a3348107ae3154e44cR10

**Comment by agstephens** [comment]
Hi @ocefpaf, The patch was needed when the cfchecker directly interacted with `udunits2`. However, the dependency has now moved to the `cfunits` package which is also part of conda-forge:

 https://github.com/conda-forge/cfunits-feedstock/

I had assumed we can depend on `cfunits` to resolve the dependency now. Is that not right? :-)

**Comment by agstephens** [comment]
@ocefpaf, does the patch need to be moved into `cfunits` - or at least a version of it moved there?

**Comment by ocefpaf** [comment]
> I had assumed we can depend on `cfunits` to resolve the dependency now. Is that not right? :-)

If the new code does not load `udunits2` this is fine. However, we may need to do something like this in the `cfunits` package then.

File: recipe/meta.yaml
Comment (ocefpaf) [inline recipe/meta.yaml]: You may take this opportunity to update this to a `sha256`.

**Comment by agstephens** [comment]
@cehbrecht: just copying you in to this conversation as it _might_ impact on the `cfunits` feedstock recipe.

File: recipe/meta.yaml
Comment (ocefpaf) [reply recipe/meta.yaml]: The build keeps failing because of the upper case package name above. See the logs:

```
BUILD START: cfchecker-3.0.5-py27_0
    (actual version deferred until further download or env creation)
Can't build /recipe_root due to unsatisfiable dependencies:
Package missing in current linux-64 channels: 
  - netCDF4
```

Comment (agstephens) [reply recipe/meta.yaml]: @ocefpaf: should be resolved. Thanks

File: recipe/meta.yaml
Comment (agstephens) [reply recipe/meta.yaml]: These two should both exist on conda-forge:
https://github.com/conda-forge/netcdf4-feedstock/
https://github.com/conda-forge/cfunits-feedstock/

@ocefpaf: I'm hoping it will pass the test now. Thanks

**Review by ocefpaf — APPROVED** [review]

**Comment by ocefpaf** [comment]
@agstephens consider adding yourself as a maintainer for this package.

File: recipe/meta.yaml
Comment (ocefpaf) [reply recipe/meta.yaml]: It looks like the entrypoint changed :unamused:

Let me take a look at this...

**Comment by agstephens** [comment]
@ocefpaf: point me to where I can make myself a maintainer. Thanks

I'm getting the hang of seeing that I check the CI output myself and fix things without hassling you. Clever!

**Comment by ocefpaf** [comment]
> @ocefpaf: point me to where I can make myself a maintainer. Thanks

All you need to do is to add your GitHub handle in the `recipe-maintainers` section.

> I'm getting the hang of seeing that I check the CI output myself and fix things without hassling you. Clever!

Take it to the next level and build it locally before pushing :wink: 

http://conda-forge.github.io/docs/testing.html#run-docker-tests-locally-for-feedstock

**Comment by ocefpaf** [comment]
@agstephens two more things before we merge this:

1) Can you rerender the feedstock? See http://conda-forge.github.io/docs/conda_smithy.html
2) It would be nice to have an actual file test. PR #4 added one, maybe we can use that or we can use a local file instead if you have a small one to commit to the repo,

**Comment by agstephens** [comment]
@ocefpaf, thanks for accepting the PR.

 I am installing conda-smithy. 

Regarding point (2) about a test file. Indeed, a local file is preferable. Does the test run inside the main directory of the repo? If so, we already have gzipped ".nc" files accessible. Can a test command include two shell commands, e.g.:
` $ gunzip test_files/data.nc.gz ; cfchecks test_files/data.nc`
Also, does it care about the return code? The checker returns a code for the number of problems it found in the file (but it still ran correctly). Thanks

**Comment by ocefpaf** [comment]
> Does the test run inside the main directory of the repo?

You need to make the test file/directory available visible to conda by adding a subsection named `files` in the `test` secton, like [this](https://github.com/conda-forge/fiona-feedstock/blob/master/recipe/meta.yaml#L46-L47)

> If so, we already have gzipped ".nc" files accessible. Can a test command include two shell commands, e.g.:
` $ gunzip test_files/data.nc.gz ; cfchecks test_files/data.nc`

You can run as many commands as you like, but I would not add unpacking as part of the test. Just add the unpackage file, see [these](https://github.com/conda-forge/fiona-feedstock/blob/master/recipe/meta.yaml#L48-L53) lines for an example.

> Also, does it care about the return code? The checker returns a code for the number of problems it found in the file (but it still ran correctly). Thanks

Anything non-zero will be treated as an error.

**Pull request merged**
